khelo/lib/ui/flow/score_board/components/revise_target_sheet.dart

```diff
@@ -0,0 +1,211 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:go_router/go_router.dart';
+import 'package:khelo/domain/extensions/context_extensions.dart';
+import 'package:khelo/ui/flow/score_board/components/bottom_sheet_wrapper.dart';
+import 'package:khelo/ui/flow/score_board/score_board_view_model.dart';
+import 'package:style/button/primary_button.dart';
+import 'package:style/extensions/context_extensions.dart';
+import 'package:style/text/app_text_field.dart';
+import 'package:style/text/app_text_style.dart';
+
+class ReviseTargetSheet extends ConsumerStatefulWidget {
+  static Future<T?> show<T>(BuildContext context) {
+    HapticFeedback.mediumImpact();
+    return showModalBottomSheet(
+      context: context,
+      showDragHandle: false,
+      enableDrag: false,
+      isScrollControlled: true,
+      useRootNavigator: true,
+      backgroundColor: context.colorScheme.surface,
+      builder: (context) {
+        return Padding(
+          padding:
+              EdgeInsets.only(bottom: MediaQuery.of(context).viewInsets.bottom),
+          child: const ReviseTargetSheet(),
+        );
+      },
+    );
+  }
+
+  const ReviseTargetSheet({super.key});
+
+  @override
+  ConsumerState createState() => _ReviseTargetSheetState();
+}
+
+class _ReviseTargetSheetState extends ConsumerState<ReviseTargetSheet> {
+  bool isButtonEnabled = false;
+  String? errorString;
+
+  TextEditingController manualOverTextController = TextEditingController();
+  TextEditingController manualRunsTextController = TextEditingController();
+
+  @override
+  Widget build(BuildContext context) {
+    final state = ref.watch(scoreBoardStateProvider);
+    return BottomSheetWrapper(
+      content: _content(context, state),
+      action: [
+        PrimaryButton(
+          context.l10n.score_board_confirm_target_text,
+          enabled: isButtonEnabled,
+          onPressed: () {
+            final runs = int.parse(manualRunsTextController.text);
+            final overs = double.parse(manualOverTextController.text);
+
+            if (_isValidInput(state, runs, overs)) {
+              context.pop({'run': runs, 'over': overs});
+            } else {
+              setState(() => errorString =
```

Comment:
I believe we should avoid using `setState` here. If we do, it will trigger a rebuild of the bottom sheet or score board screen, which is inefficient. Let's handle all logic within the view model instead.